core/validation/validate.py

```diff
@@ -141,7 +142,10 @@ def validate_types(data_dict: dict[str, pd.DataFrame], table: str, column_to_typ
     for index, row in data_dict[table].iterrows():
         for column, exp_type in column_to_type.items():
             got_value = row.get(column)
-            if got_value is None or pd.isna(got_value):
+
+            if (isinstance(got_value, list) and got_value == []) or (
+                not isinstance(got_value, list) and got_value in NA_VALUES
+            ):
```

Comment:
The Postcodes column could potentially be something in ```NA_VALUES``` - if it's empty it won't necessarily be instantiated as an empty list, it could be treated as a null value. Better to do simply do ```if got_value in NA_VALUES or not got_value```.